core/src/main/java/bio/terra/pearl/core/dao/i18n/LanguageTextDao.java

```diff
@@ -29,14 +29,26 @@ public void deleteByKeyNameAndPortal(String keyName, UUID portalId) {
         deleteByTwoProperties("key_name", keyName, "portal_id", portalId);
     }
 
-    //returns all language texts that are either global or portal specific
-    public List<LanguageText> findByPortalIdOrNullPortalId(UUID portalId, String language) {
+    //returns all language texts for a portal environment, with overrides for the given language
+    public List<LanguageText> findWithOverridesByPortalEnvId(UUID portalEnvId, String language) {
         return jdbi.withHandle(
             handle ->
                 handle
                     .createQuery(
-                        "SELECT * FROM language_text WHERE (portal_id = :portalId OR portal_id IS NULL) AND language = :language")
-                    .bind("portalId", portalId)
+                            // distinct on key_name and order by localized_site_content_id is used to ensure that
+                            // if there are conflicting keys, the one with localized_site_content_id will be used
+                            """
+                                    SELECT DISTINCT ON(lt.key_name) lt.* FROM language_text lt
+                                    LEFT JOIN localized_site_content lsc ON lsc.id = lt.localized_site_content_id
+                                    LEFT JOIN site_content sc ON sc.id = lsc.site_content_id
```

Comment:
I don't think you need/want either of these joins.  I think you made the correct decision to use subqueries instead